### PR TITLE
Searchable Callouts & Whiteboard text

### PR DESCRIPTION
**Pull request by hero101**
## Search on
- Callouts
- Whiteboard text (ONLY the text found in the whiteboards) - The assosiated Calllout will be returned instead of the Whiteboard

## Ingest
- data is ingested in the background. A task is returned instead.

### CODE REVIEW

File: .env.docker [hunk 1]
```diff
@@ -5,8 +5,8 @@ MYSQL_DATABASE=alkemio
 MYSQL_ROOT_PASSWORD=toor
 DATABASE_HOST=mysql
 
-SYNAPSE_HOMESERVER_NAME=alkemio.matrix.host
-SYNAPSE_SERVER_NAME=alkemio.matrix.host
+SYNAPSE_HOMESERVER_NAME=matrix-acc.alkem.io
```
Comment (valentinyanakiev) [inline .env.docker]: these settings need to be reverted

File: src/services/api/search/v2/extract/search.extract.service.ts [hunk 2]
```diff
@@ -32,6 +30,9 @@ const TYPE_TO_INDEX = (
   [SearchEntityTypes.POST]: `${indexPattern}posts`,
   [SearchEntityTypes.USER]: `${indexPattern}users`,
   [SearchEntityTypes.ORGANIZATION]: `${indexPattern}organizations`,
+  [SearchEntityTypes.CALLOUT]: `${indexPattern}callouts`,
+  [SearchEntityTypes.GROUP]: '',
```
Comment (valentinyanakiev) [inline src/services/api/search/v2/extract/search.extract.service.ts:34]: Is this needed?

File: src/services/api/search/search.entity.types.ts [hunk 3]
```diff
@@ -1,7 +1,10 @@
 export enum SearchEntityTypes {
   USER = 'user',
+  GROUP = 'group', // may be removed
```
Comment (valentinyanakiev) [inline src/services/api/search/search.entity.types.ts:3]: Is this needed?

File: src/platform/admin/search/admin.search.ingest.resolver.mutations.ts [hunk 4]
```diff
@@ -41,27 +48,67 @@ export class AdminSearchIngestResolverMutations {
...
+          );
+        }
+        return this.searchIngestService.ingest(task);
+      })
+      .then(() => this.taskService.complete(task.id))
+      .catch(async e => {
+        await this.taskService.updateTaskErrors(task.id, e?.message);
+        return this.taskService.complete(task.id, TaskStatus.ERRORED);
+      });
+    // const deleteResult = await this.searchIngestService.removeIndices();
```
Comment (valentinyanakiev) [inline src/platform/admin/search/admin.search.ingest.resolver.mutations.ts]: Cleanup / raise story.

File: src/platform/admin/search/admin.search.ingest.resolver.mutations.ts [hunk 5]
```diff
@@ -41,27 +48,67 @@ export class AdminSearchIngestResolverMutations {
...
-      throw new BaseException(
-        deleteResult.message ?? 'Failed to delete indices',
-        LogContext.SEARCH_INGEST,
-        AlkemioErrorStatus.UNSPECIFIED
-      );
-    }
+    this.logger.verbose?.('Starting search ingest from scratch');
+
+    const task = await this.taskService.create();
+    // todo: add error handling
```
Comment (valentinyanakiev) [inline src/platform/admin/search/admin.search.ingest.resolver.mutations.ts]: I see error handling, is this toDo still valid?

File: src/platform/admin/search/admin.search.ingest.resolver.mutations.ts [hunk 6]
```diff
@@ -15,19 +15,26 @@ import {
...
     private platformAuthorizationPolicyService: PlatformAuthorizationPolicyService,
-    private searchIngestService: SearchIngestService
+    private searchIngestService: SearchIngestService,
+    private taskService: TaskService,
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private logger: LoggerService
   ) {}
 
   @UseGuards(GraphqlGuard)
-  @Mutation(() => AdminSearchIngestResult, {
+  // @Mutation(() => AdminSearchIngestResult, {
```
Comment (valentinyanakiev) [inline src/platform/admin/search/admin.search.ingest.resolver.mutations.ts]: Why is this type changed?

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]

File: src/platform/admin/search/admin.search.ingest.resolver.mutations.ts (hunk 6 above)
Comment (hero101) [reply src/platform/admin/search/admin.search.ingest.resolver.mutations.ts]: It now returns the id of the task running in the background

File: src/platform/admin/search/admin.search.ingest.resolver.mutations.ts (hunk 4 above)
Comment (hero101) [reply src/platform/admin/search/admin.search.ingest.resolver.mutations.ts]: No longer needed. The progress is handled in the task.

File: src/services/api/search/search.entity.types.ts (hunk 3 above)
Comment (hero101) [reply src/services/api/search/search.entity.types.ts:3]: Yes. There were a couple of duplicate types, so I've created one complete type.
The group is still needed in search v1.

File: src/services/api/search/v2/extract/search.extract.service.ts (hunk 2 above)
Comment (hero101) [reply src/services/api/search/v2/extract/search.extract.service.ts:34]: Yes. There were a couple of duplicate types, so I've created one complete type.
The group is still needed in search v1.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**